L13-Python-Interceptors.md

```diff
@@ -0,0 +1,285 @@
+
+
+## Background
+
+### The Need for Interceptors
+
+Interceptors are middleware which capable of manipulating or reacting to the transporting requests/responses throughout the life cycle of an RPC. It dramatically simplifies the implementation of tracing (See [grpc/grpc#8767](https://github.com/grpc/grpc/issues/8767)) and empowers gRPC developers to add general channel-level business logic.
+
+Interceptors have existed with gRPC for a while now, and its semantics were carried out across gRPC implementations like Java, Go and wrapper languages like Node.js, Ruby.
+
+### Stabilization Status
+
+| gRPC Language | Client Interceptor | Server Interceptor |
+| ------------- | ------------------ | ------------------ |
+| Python        | Experimental       | Experimental       |
+| Node.js       | Stable             | N/A                |
+| Java          | Stable             | Stable             |
+| Golang        | Experimental       | Experimental       |
+| C++           | N/A                | N/A                |
+| Ruby          | Experimental       | Experimental       |
+
+### Related Proposals
+
+* [Node.js Client Interceptor](https://github.com/grpc/proposal/blob/master/L5-node-client-interceptors.md)
+* [Ruby Client and Server Interceptors](https://github.com/grpc/proposal/pull/34)
+* [C# Client and Server Interceptors](https://github.com/grpc/proposal/pull/38)
+
+
+## Proposal
+
+### Interface: Client Call Details
+
+To intercept the RPC, gRPC needs to describe the upcoming RPC that enables the interceptor function to inspect and modify the call options. Hence we introduce the new interface `grpc.ClientCallDetails`, which contains five properties:
+
+```Python
+class ClientCallDetails(six.with_metaclass(abc.ABCMeta)):
+    """Describes an RPC to be invoked.
+
+    Attributes:
+      method: The method name of the RPC.
+      timeout: An optional duration of time in seconds to allow for the RPC.
+      metadata: Optional metadata to be transmitted to the service-side of the RPC.
+      credentials: An optional CallCredentials for the RPC.
+      wait_for_ready: This is an EXPERIMENTAL argument. An optional flag to enable wait for ready mechanism.
+    """
+```
+
+### Concept: Continuation
+
+The `continuation` function is a function that proceeds with the invocation by executing the next interceptor in the chain or invoking the actual RPC on the underlying Channel. It is the interceptor’s responsibility to call it if it decides to move the RPC forward. The interceptor can use `response_future = continuation(client_call_details, request)` to continue with the RPC. Continuation returns an object that is both a Call for the RPC and a Future. In the event of RPC completion, the return Call-Future’s result value will be the response message of the RPC. Should the event terminate with non-OK status, the returned Call-Future’s exception value will be an RpcError.
```

Comment:
"responsibility to call it" -> "responsibility to call the continuation"

Reply:
"if it decides to move the RPC forward" -> "unless it decides to terminate processing of the RPC"?

Reply:
"Continuation returns an object" -> "The continuation returns an object"

Reply:
Updated and marked the word "continuation" with "`".